server/src/main/java/io/crate/auth/AccessControlImpl.java

```diff
@@ -759,13 +759,15 @@ public Void visitCreatePublication(AnalyzedCreatePublication createPublication,
             // and a table gets published only if publication owner has all privileges on that table.
             for (RelationName relationName: createPublication.tables()) {
                 for (Privilege.Type type: VALUES) {
-                    Privileges.ensureUserHasPrivilege(
-                        type,
-                        Privilege.Clazz.TABLE,
-                        relationName.fqn(),
-                        user,
-                        defaultSchema
-                    );
+                    if (!type.equals(Privilege.Type.AL)) {
```

Comment:
I'd prefer making this more concrete to avoid issues when more types would be added. 
We could store DQL,DML,DDL inside as static list, e.g. DATA_TYPES (not perfect, maybe you have a better naming idea), at the Privilege class